### PR TITLE
Reformat ES6 Challenge Descriptions

### PR DESCRIPTION
**Pull request by robbawebba**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.
- [ ] Closes currently open issue (replace XXXX with an issue no): Closes #XXXX
Progress towards issue #12698 

#### Description

Reformat descriptions, titles and challenge seed when necessary for 4 Challenges in ES6 section: Nested Object Destructuring, Array Destructuring, Array Destructuring with Rest Operator, and Destructuring Function Parameters

### CODE REVIEW

File: seed/challenges/02-javascript-algorithms-and-data-structures/es6.json [hunk 1]
```diff
@@ -471,23 +468,21 @@
...
+        "Variables <code>a</code> and <code>b</code> take the first and second values from the array. After that, because of rest operator's presence, <code>arr</code> gets rest of the values in the form of an array.",
+        "The rest element only works correctly as the last variable in the list. As in, you cannot use the rest operator to catch a subarray that leaves out last element of the original array.",
+        "<hr>",
+        "Use destructuring assignment with the rest operator to perform an effective <code>Array.prototype.slice()</code> so that <code>arr</code> is a sub-array of the original array <code>source</code> with the first two elements ommitted."
       ],
       "challengeSeed": [
         "const source = [1,2,3,4,5,6,7,8,9,10];",
-        "/* Alter code below this line */",
+        "// change code below this line",
         "const arr = source ; // change this",
```
Comment (Greenheart) [inline seed/challenges/02-javascript-algorithms-and-data-structures/es6.json]: We might want to remove this space for consistency.

File: seed/challenges/02-javascript-algorithms-and-data-structures/es6.json [hunk 2]
```diff
@@ -527,9 +519,9 @@
         "    min: -0.75,",
         "    average: 35.85",
         "}",
```
Comment (Greenheart) [inline seed/challenges/02-javascript-algorithms-and-data-structures/es6.json]: Missing semicolon here.

**Review by Greenheart — CHANGES_REQUESTED** [review]
@robbawebba This is a big improvement, great job! :smile:

Just a few minor thoughts below.

**Comment by robbawebba** [comment]
Thanks for the suggestions @Greenheart!

**Review by Greenheart — APPROVED** [review]
@robbawebba Thanks for fixing this! :smile:

**Pull request merged**